src/test/sushi-integration.t.sol

```diff
@@ -627,93 +671,159 @@ contract SushiIntegrationTest is TestBase {
+        assertEq(masterchef.owner(), address(user2));
+    }
+
+    function test_cage_queued_dangerous_change_owner2() public {
+        execute_dangerous_timelock_action("", abi.encodeWithSelector(MasterChefLike.transferOwnership.selector, address(user2), true, false));
+        assertEq(masterchef.owner(), address(user2));
+    }
+
+    function test_cage_queued_dangerous_change_rewarder1() public {
+        execute_dangerous_timelock_action("set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(user2), true));
+        assertEq(masterchef.rewarder(join.pid()), address(user2));
+    }
+
+    function test_cage_queued_dangerous_change_rewarder2() public {
+        execute_dangerous_timelock_action("", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), true));
+        assertEq(masterchef.rewarder(join.pid()), address(user2));
+    }
+
+    function test_cage_queued_safe_change_diff_target() public {
+        queue_safe_timelock_action(address(user1), "setMigrator(address)", abi.encode(address(user2)), false);
+    }
+
+    function test_cage_queued_safe_change_safe_function1() public {
+        queue_safe_timelock_action(address(masterchef), "add(uint256,address,address)", abi.encode(100, address(pair), address(user2)), true);
+    }
+
+    function test_cage_queued_safe_change_safe_function2() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.add.selector, 100, address(pair), address(user2)), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args1() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(user2), false), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args2() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(user2), false), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args3() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid(), 100, address(rewarder), true), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args4() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid(), 100, address(rewarder), true), true);
+    }
+
+    function test_cage_queued_safe_change_safe_args5() public {
+        queue_safe_timelock_action(address(masterchef), "set(uint256,uint256,address,bool)", abi.encode(join.pid() + 1, 100, address(user2), true), false);
+    }
+
+    function test_cage_queued_safe_change_safe_args6() public {
+        queue_safe_timelock_action(address(masterchef), "", abi.encodeWithSelector(MasterChefLike.set.selector, join.pid() + 1, 100, address(user2), true), false);
+    }
```

Comment:
I love the thoroughness here, but could we go a step further and give the tests either more specific names or comments indicating why the change is "safe"?

Reply:
Also it seems in some invocations you're passing `execute` as `true` and in others, `false`. IIUC it could be false in all of these calls as all we care about is ensuring the adapter can't be `caged`. More generally I suspect you could factor+name the `queue`/`cage`/`execute` helpers more cleanly if you were so inclined--e.g. `queue_safe_timelock_action` is confusingly named because it also attempts to `cage` and optionally `executes`. I might take an approach like defining a `TimeLockTX` struct to make passing around the data easy, and then have simpler, more single-purpose functions like `queue_timelock_tx(TimeLockTX tx_info)`, `try_cage(TimeLockTX tx_info, bool expectSuccess)` and `execute_timelock_tx(TimeLockTX tx_info)`. Expressing the tests in terms of these might be more lines, but IMO would be far more readable and maintainable.

Reply:
I'll add some comments to explain why things are safe and not safe. The reason I want to attempt to execute if available is because of this bug that almost made it through: https://github.com/makerdao/dss-crop-join/blob/master/src/sushi.sol#L33

As you can see that interface for `setMigrator` was the wrong interface. If I had execution on then the timelock execute would fail because it's an invalid call. It's just a double check the interfaces are correct.

Where I set the safe action to not execute, it's just an invalid call. Actually since more pools have been added `test_cage_queued_safe_change_safe_args5/6` is now valid, so I've turned those on. I get this could be prettied up a bit, but I think it works fine and I don't really want to spend too much time on it for cosmetic reasons. Maybe if I get some spare time.